swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -53,6 +53,7 @@ final class IterableAPIInternal : NSObject, PushTrackerProtocol {
             _userId = newValue
             _email = nil
             storeEmailAndUserId()
+            createUserForUserId(onSuccess: IterableAPIInternal.defaultOnSucess(identifier: "createUserForUserId"), onFailure: IterableAPIInternal.defaultOnFailure(identifier: "createUserForUserId"))
             
```

Comment:
We should probably wait for the call to createUserForUserId to finish before calling enableDevice. That way we know that the user exists before enabling the device. Also adding a unit test would be good.